### PR TITLE
[ios] Test scripts for bundling map files with framework

### PR DESCRIPTION
**Pull request by julianrex**
WIP experiment, enabling `LD_GENERATE_MAP_FILE` for the dynamic and static targets.

@friedbunny I've attempted to modify `package.sh` - but scripting is not my forte. Would you mind taking a look?

### CODE REVIEW

File: platform/ios/scripts/package.sh [hunk 1]
```diff
@@ -98,6 +99,17 @@ fi
 
 LIBS=(Mapbox.a)
 
+function copy_map_files {
```
Comment (friedbunny) [inline platform/ios/scripts/package.sh]: Having this as a named function is great if it’s called more than once, but I think doing this once would be possible if it were moved down to the `Copying library resources` section (where the bcsymbolmaps and such are copied).

File: platform/ios/scripts/package.sh [hunk 2]
```diff
@@ -98,6 +99,17 @@ fi
...
+    MAP_FILE_ARCHS='armv7 arm64'
+    for ARCH in ${MAP_FILE_ARCHS}; do
+        MAP_FILE="${PRODUCTS}/${BUILDTYPE}-iphoneos/${NAME}-LinkMap-${VARIANT}-${ARCH}.txt"
+        if [[ -e ${MAP_FILE} ]]; then
+            step "Copying map file for ${ARCH}"
+            cp ${MAP_FILE} ${OUTPUT}/dynamic/
```
Comment (friedbunny) [inline platform/ios/scripts/package.sh]: I like to add the `-v` flag, so that file names and paths are preserved in logs.

**Comment by julianrex** [comment]
> This seems useful for us, but are they useful for end-developers? 

That's a good question. dSYMs should be sufficient; these are more for us - I feel they should be kept with the other release artifacts - if the framework & dsyms are also there in the CircleCI job, then sounds like that would be a good spot.

**Comment by julianrex** [comment]
Still on my list.

**Comment by julianrex** [comment]
Please reopen.

**Pull request closed without merging**